src/App/component/IssuesPage.jsx

```diff
@@ -100,32 +100,56 @@ function IssuesPage(prop) {
   }, [currentProject, prop.startMonth, prop.endMonth])
 
   useEffect(() => {
-    const { endMonth } = prop
-    let chartDataset = { labels: [], data: { created: [], closed: [] } }
-    let issueListDataSortedByCreatedAt = issueListData
-    let issueListDataSortedByClosedAt = issueListData
+    const chartDataset = generateIssueDataSet()
+    setDataForIssueChart(chartDataset)
+  }, [issueListData])
 
-    issueListDataSortedByCreatedAt.sort((a, b) => a.createdAt - b.createdAt)
-    issueListDataSortedByClosedAt.sort((a, b) => a.closedAt - b.closedAt)
+  const generateIssueDataSet = () => {
+    const { startMonth, endMonth } = prop
+    let chartDataset = { labels: [], data: { closed: [] , created: []} }
+    for (let month = moment(startMonth); month <= moment(endMonth); month = month.add(1, 'months')) {
+      chartDataset.labels.push(month.format("YYYY-MM"))
+    }
+    chartDataset.data.created = getIssueCreatedCount()
+    chartDataset.data.closed = getIssueClosedCount()
 
-    if (issueListDataSortedByCreatedAt.length > 0) {
-      for (let month = moment(issueListDataSortedByCreatedAt[0].createdAt); month <= moment(endMonth).add(1, 'months'); month = month.add(1, 'months')) {
-        let index
-        chartDataset.labels.push(month.format("YYYY-MM"))
+    return chartDataset
+  }
 
-        index = issueListDataSortedByCreatedAt.findIndex(issue => {
+  const getIssueCreatedCount = () => {
+    const { startMonth, endMonth } = prop
+    let created = []
+    let issueListDataSortedByCreatedAt = issueListData
```

Comment:
Suggest make these variables to be constants.
```suggestion
    const created = []
    const issueListDataSortedByCreatedAt = issueListData
```